core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1542,4 +1542,50 @@ package object config {
     .bytesConf(ByteUnit.BYTE)
     .createOptional
 
+  private[spark] val GRACEFUL_DECOMMISSION_ENABLE =
```

Comment:
Would it make sense to handle this with `spark.worker.decommission.enabled` or not? I'm not sure just for discussion.

Reply:
is this specific to yarn or been tied in with previous work for all cluster managers?

Reply:
also all the configs need to be documented in configuration.md